proxy/src/coco/config.rs

```diff
@@ -82,3 +82,72 @@ pub fn configure(
         disco,
     }
 }
+
+/// Resolve seed identifiers into `(PeerId, SocketAddr)` pairs.
+///
+/// The expected format is `<peer-id>@<host>:<port>`
+///
+/// # Errors
+///
+/// If any of the supplied seeds cannot be parsed, an error is returned.
+pub async fn resolve_seeds<T: AsRef<str> + Send + Sync>(
```

Comment:
Hmm wouldn't it be a `Seeds` struct? Ie. if we want to avoid having `resolve_seeds`. Otherwise we can keep this function and call `Seed::from` for each seed..

Reply:
Ah so the `str` is the new-line separated seeds? I would probably break it down into `FromStr` for a single `Seed` then maybe have a `parse_seeds: AsRef<str> -> Result<Vec<Seed>, Error>` where it splits on new-line and parses individual lines.

Reply:
`FromStr` won't work because the function is async. I can implement a resolve method though on `Seed`

Reply:
I see, because you need to look up the host. So maybe it should be:
```rust
struct Seed<Host> {
  peer_id: PeerId,
  host: Host,
}

impl FromStr for Seed<String> { /* split and parse */ }

impl Seed {
  pub fn resolve(&self) -> Seed<PeerId, SocketAddr>
}
```

What do you think?

Reply:
We could do that, but I'm not sure what we gain. These functions are used internally and only in one place at the moment.

Reply:
We'd gain the separation of each stage:
* Parse a string into peer and host
* Resolve the host
* And now we have a fully parsed and fully resolved seed :)